src/Commands/Site/InfoCommand.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Site;
+
+use Consolidation\OutputFormatters\StructuredData\AssociativeList;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+
+class InfoCommand extends TerminusCommand
+{
+    /**
+     * Gets full site information
+     *
+     * @command site:info
+     *
+     * @field-labels
+     *   id: ID
+     *   name: Name
+     *   label: Label
+     *   created: Created
+     *   framework: Framework
+     *   organization: Organization
+     *   service_level: Service Level
+     *   upstream: Upstream
+     *   php_version: PHP Version
+     *   holder_type: Holder Type
+     *   holder_id: Holder ID
+     *   owner: Owner
+     * @param string $site Name|UUID of a site to look up
+     * @option field Individual field to return from requested site
+     * @usage terminus site:info <site>
+     *   * Responds with the table view of site information
+     *   * Responds that you are forbidden if you access a site that exists
+     *      but you do not have access to it
+     *   * Responds that a site does not exist
+     * @usage terminus site:info --field=<field>
+     *   * Responds with the single field of site information requested
+     *   * Responds that you are forbidden if you access a site that exists
+     *      but you do not have access to it
+     *   * Responds that a site does not exist
+     * @return AssociativeList
+     */
+    public function lookup($site, $options = ['field' => null])
+    {
+        $sites = new Sites();
+
+        $response = $sites->get($site)->serialize();
+        // Hopefully temporary fix for outputting the upstream (which is an array)
+        foreach ($response as &$value) {
+            if (is_array($value)) {
+                // If it's null (null means full table view), use newline, otherwise use commas
+                $value = is_null($options['field']) ? implode("\n", $value) : implode(", ", $value);
```

Comment:
This isn't a good test, as there are other ways to set the format to something else -- e.g. --format=yaml or --format=string.  This could should either **invariantly** flatten, or else it should not flatten at all, and should instead return a subclass of AssociativeList that implements [RenderCellInterface](https://github.com/consolidation/output-formatters/blob/master/src/StructuredData/RenderCellInterface.php).  If you do that, then `renderCell()` will be called in any instance where the array should be flattened to a string.  Be sure to test the `$key` and data type of the cell data as necessary, and pass through any data that you don't want to flatten or otherwise transform.  The advantage of that is that your array will remain an array in yaml and json format, et. al., which might be relevant to the requestor.